reolink_aio/api.py

```diff
@@ -3222,6 +3222,194 @@ async def reboot(self, channel: int | None = None) -> None:
+            self.nvr_name,
+            cert_size,
+            key_size,
+        )
+
+        try:
+            json_data = await self.send(body, expected_response_type="json")
+
+            if json_data[0]["code"] != 0 or json_data[0].get("value", {}).get("rspCode", -1) != 200:
+                err_mess: dict = json_data[0].get("error", {})
+                rspCode = json_data[0].get("value", err_mess).get("rspCode", -1)
+                detail = json_data[0].get("value", err_mess).get("detail", "")
+                raise ApiError(
+                    f"ImportCertificate: API returned error code {json_data[0]['code']}, response code {rspCode}/{detail}",
+                    rspCode=rspCode,
+                )
+
+            _LOGGER.info("Successfully imported certificate to %s", self.nvr_name)
+            return True
+
+        except ReolinkError as err:
+            _LOGGER.error("Error importing certificate to %s: %s", self.nvr_name, err)
+            raise
+
+    async def upload_certificate(
+        self,
+        cert_content: str,
+        key_content: str,
+        cert_name: str = "server",
+        relogin_delay: float = 5.0,
+    ) -> bool:
+        """
+        Upload SSL/TLS certificate to the device (complete workflow).
+
+        This method handles the complete certificate upload workflow:
+        1. Clear existing certificates
+        2. Wait and re-login (required after clearing)
+        3. Import new certificate
+        4. Wait for certificate to be processed
+
+        Args:
+            cert_content: PEM-encoded certificate content.
+            key_content: PEM-encoded private key content.
+            cert_name: Base name for certificate files (default: "server").
+            relogin_delay: Seconds to wait before re-login after clearing certificates (default: 5.0).
+
+        Returns:
+            True if upload succeeded.
+
+        Raises:
+            InvalidParameterError: If certificate or key content is invalid.
+            ApiError: If the API returns an error.
+            LoginError: If re-login after clearing certificates fails.
+
+        Note:
+            - The device only supports RSA certificates, not EC certificates.
+            - Some models (e.g., E1 Pro) may require a reboot to activate the certificate.
+            - The workflow requires re-authentication after clearing certificates.
+        """
+        _LOGGER.info("Starting certificate upload workflow for %s", self.nvr_name)
+
+        # Step 1: Clear existing certificates
+        await self.clear_certificates()
+
+        # Step 2: Wait and re-login (required after clearing certificates)
+        _LOGGER.info("Waiting %s seconds before re-login...", relogin_delay)
+        await asyncio.sleep(relogin_delay)
+
+        # Expire current session and force re-login
+        await self.expire_session(unsubscribe=False)
+
+        # Force a command that requires login to trigger re-authentication
```

Comment:
Why is this needed here?

It sounds more logical to first import the new certificate???
Why do you need to first logout and login after clearing the certificate, also are you even able to login after clearing the certificate?

Reply:
I tested it a bit, I don't know if I did it correctly. --- and discovered that my camera basically applies the cert, which means that the next request sees the applied cert. But it looks like the cam did not need a relogin ---
I think it might be needed, but on the other hand, it does not harm. I would leave it in. What is your opinion?